js/test/test/testingSupport.js

```diff
@@ -0,0 +1,814 @@
+/*
+ * testingSupport.js - random utilities to use only in the unit tests
+ *
+ * Copyright © 2025, JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+if (typeof(SearchUtils) === 'undefined') {
+    var SearchUtils = require("../../lib/SearchUtils");
+}
+if (typeof(semver) === 'undefined') {
+    var semver = require("semver");
+}
+if (typeof(ilib) === 'undefined') {
+    var ilib = require("../../lib/ilib.js");
+}
+
+// ICU Version, CLDR Version, Unicode Version
+var icuCldrVersionMappings = [
+    [ 78, 48.0, 16.0 ], // planned for 2026
+    [ 77.1, 47.0, 16.0 ],
+    [ 77, 47.0, 16.0 ],
+    [ 76.1, 46.0, 16.0 ],
+    [ 76, 46.0, 16.0 ],
+    [ 75.1, 45.0, 15.1 ],
+    [ 75, 45.0, 15.1 ],
+    [ 74.2, 44.1, 15.1 ],
+    [ 74, 44.1, 15.1 ],
+    [ 73, 45.0, 15.0 ],
+    [ 72, 42.0, 15.0 ],
+    [ 71.1, 41.0, 14.0 ],
+    [ 71, 41.0, 14.0 ],
+    [ 70.1, 40.0, 14.0 ],
+    [ 70, 40.0, 14.0 ],
+    [ 69, 39.0, 14.0 ],
+    [ 68, 38.0, 13.0 ],
+    [ 67.1, 37.0, 13.0 ],
+    [ 67, 37.0, 13.0 ],
+    [ 66, 36.1, 13.0 ],
+    [ 65, 35.1, 12.1 ],
+    [ 64.2, 35.1, "" ],
+    [ 64, 34.0, 12.0 ],
+    [ 63.1, 34.0, 12.0 ],
+    [ 63, 33.1, 11.1 ],
+    [ 62, 32.0, 11.0 ],
+    [ 61, 31.0, 10.0 ],
+    [ 60, 30.0, 9.0 ],
+    [ 59, 29.0, 9.0 ],
+    [ 58, 28.0, 9.0 ]
+];
+
+// format is:
+// Chrome Version (Approx.), ICU Version, Release Date (Approx.)
+var chromeVersionMappings = [
+    [ 142, 76, "Nov 2025" ],
```

Comment:
where did you get these information from? Last time I check, It wasn't easy to figure out the specific ICU version used by a particular chromium build. I had to look up chromium's git, find the hash value, and the locate the commit for that value.. ;;;